data/hooks/conf_regen/34-mysql

```diff
@@ -1,58 +1,41 @@
 #!/bin/bash
 
 set -e
-MYSQL_PKG="$(dpkg --list | sed -ne 's/^ii  \(mariadb-server-[[:digit:].]\+\) .*$/\1/p')"
 . /usr/share/yunohost/helpers
 
 do_pre_regen() {
   pending_dir=$1
 
   cd /usr/share/yunohost/templates/mysql
 
+  # Legacy code to get rid of /etc/yunohost/mysql ...
+  # Nowadays, we can simply run mysql while being run as root of unix_socket/auth_socket is enabled...
+  if [ -f /etc/yunohost/mysql ]; then
+
+      # This is a trick to check if we're able to use mysql without password
+      # Expect instances installed in stretch to already have unix_socket
+      #configured, but not old instances from the jessie/wheezy era
+      if ! echo "" | mysql
+      then
+          password="$(cat /etc/yunohost/mysql)"
+          # Enable plugin unix_socket for root on localhost
+          mysql -u root -p"$password" <<< "GRANT ALL PRIVILEGES ON *.* TO 'root'@'localhost' IDENTIFIED WITH unix_socket WITH GRANT OPTION;"
+      fi
+
+      # If now we're able to login without password, drop the mysql password
+      if echo "" | mysql
+      then
+          rm /etc/yunohost/mysql
+      fi
```

Comment:
Maybe we can manage more this situation. Because in this case at each regen-conf, we will try to update the privilege in mysql and don't really make sens as if the first time it fail the other time it will probably also fail...

Reply:
Hmyea I guess at least we can display a warning ...

I don't know much about what else to do though ... If enabling the auth_socket plugin doesn't work, that means that either the mysql didn't work, or that there's some issue with the structure or permission of the mysql db ... In both cases, that means there are bigger issues on the system and we can't fix it automagically. Yet, we do need this if we expect to be able to run mysql without password in other parts of yunohost like the mysql helpers ...

Reply:
OK, Yes,

We can just add a warning.